### PR TITLE
intl: reorganize tests, add coverage

### PR DESCRIPTION
**Pull request by ryzokuken**
some more intl fixes, better test organization and increase coverage

### CODE REVIEW

File: polyfill/test/intl.mjs [hunk 1]
```diff
@@ -4,14 +4,16 @@ const { describe, it, report } = Demitasse;
 import Pretty from '@pipobscure/demitasse-pretty';
 const { reporter } = Pretty;
 
-import { strict as assert } from 'assert';
-const { deepEqual, equal, throws } = assert;
+import { strict as Assert } from 'assert';
+const { deepEqual, equal, ok: assert, throws } = Assert;
```
Comment (Ms2ger) [inline polyfill/test/intl.mjs:8]: I think you'll want to revert this.

**Comment by ryzokuken** [comment]
Converted to draft PR so that someone doesn't accidentally merge this, it is rebased on top of #496 in order to get a better idea of the actual coverage changes.

File: polyfill/test/intl.mjs (hunk 1 above)
Comment (ryzokuken) [reply polyfill/test/intl.mjs:8]: I don't understand, what's wrong here?

**Comment by ryzokuken** [comment]
Now this PR is ready for review/merge again. There's no change in coverage, because intl.mjs is already at 100% coverage (yay :tada:).

I did remove half the tests (I'm still testing for both English and German, just not together) for efficiency and readability, since it didn't affect coverage. If any of you insist, I'd test twice for each method again, but you may merge it as-is.

**Comment by ptomato** [comment]
I don't mind this as-is, but I'd prefer continuing to test both English and German, for completeness that isn't expressed by the coverage number.

File: polyfill/test/intl.mjs (hunk 1 above)
Comment (Ms2ger) [reply polyfill/test/intl.mjs:8]: We just made them all consistent in #448, why should this one be inconsistent?

Comment (Ms2ger) [inline polyfill/test/intl.mjs:8]:
```suggestion
import { strict as assert } from 'assert';
const { deepEqual, equal, throws } = assert;
```

**Pull request merged**